src/Analysis/Ast/Impl/Extensions/AnalysisExtensions.cs

```diff
@@ -44,9 +44,7 @@ public static IScope FindScope(this IDocumentAnalysis analysis, SourceLocation l
             var f = analysis.GetOrCreateFunction(name);
             if (f != null) {
                 foreach (var o in f.Overloads.OfType<PythonFunctionOverload>()) {
-                    if(parameters != null) {
-                        o.SetParameters(parameters);
-                    }
+                    o.SetParameters(parameters ?? new List<ParameterInfo>());
```

Comment:
We typically use something like `Array.Empty` or `Enumerable.Empty` instead of creating an empty list logic.

Also, setting an empty list compared to not setting it at all seems like a difference; especially if we move to the idea we talked about (determining if the parameters were set at all, compared to actually empty).

Reply:
I felt that having the null check there was uglier, but if the plan is to separate the idea of having no parameters or having none set, then I'll switch it out.